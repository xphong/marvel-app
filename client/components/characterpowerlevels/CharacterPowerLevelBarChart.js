import React from 'react';
import PropTypes from 'prop-types';

import { Bar } from 'react-chartjs-2';
import WithChart from '../hocs/WithChart';

const BarChart = WithChart(Bar);

export const CharacterPowerLevelBarChart = props => {
  const { characterChartData } = props;

  return (
    <div>
      <BarChart data={characterChartData} />
    </div>
  );
};

CharacterPowerLevelBarChart.propTypes = {
  characterChartData: PropTypes.object.isRequired
};

export default CharacterPowerLevelBarChart;
