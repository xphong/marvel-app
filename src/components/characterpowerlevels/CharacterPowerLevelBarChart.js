import React, { Component, PropTypes } from 'react';

import { Bar as Bar } from 'react-chartjs';

export default class CharacterPowerLevelBarChart extends Component {
  constructor(props){
    super(props);
  }

  render() {
    const {characterChartData} = this.props;

    return (
      <Bar className='chart' data={characterChartData} width='350' height='350'/>
    );
  }
}

CharacterPowerLevelBarChart.propTypes = {
  characterChartData: PropTypes.object.isRequired
};
