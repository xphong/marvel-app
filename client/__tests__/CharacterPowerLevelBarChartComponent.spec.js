import React from 'react';
import ReactDOM from 'react-dom';

import CharacterPowerLevelBarChart from '../components/characterpowerlevels/CharacterPowerLevelBarChart';

describe('CharacterPowerLevelBarChart Component', () => {
  it('should render CharacterPowerLevelBarChart', () => {
    const props = {
      characterChartData: {
        label: 'Character data set'
      }
    };

    const wrapper = shallow(<CharacterPowerLevelBarChart {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
