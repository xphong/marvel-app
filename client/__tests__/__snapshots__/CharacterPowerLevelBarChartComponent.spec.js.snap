// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CharacterPowerLevelBarChart Component should render CharacterPowerLevelBarChart 1`] = `
<div>
  <Bar
    className="chart"
    data={
      Object {
        "label": "Character data set",
      }
    }
    height={350}
    options={
      Object {
        "legend": Object {
          "display": false,
        },
        "maintainAspectRatio": false,
      }
    }
    width={350}
  />
</div>
`;
